packages/mds-ingest-service/repository/migrations/1630418242374-FixMigrationModelDiscrepancies.ts

```diff
@@ -0,0 +1,21 @@
+import { MigrationInterface, QueryRunner } from 'typeorm'
+
+export class FixMigrationModelDiscrepancies1630418242374 implements MigrationInterface {
+  name = 'FixMigrationModelDiscrepancies1630418242374'
+
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(`ALTER TABLE "event_annotations" ALTER COLUMN "events_row_id" SET NOT NULL`)
+    await queryRunner.query(
+      `ALTER TABLE "event_annotations" ADD CONSTRAINT "uc_events_row_id_event_annotations" UNIQUE ("events_row_id")`
+    )
+    await queryRunner.query(
+      `ALTER TABLE "event_annotations" ADD CONSTRAINT "fk_events_id" FOREIGN KEY ("events_row_id") REFERENCES "events"("id") ON DELETE NO ACTION ON UPDATE NO ACTION`
```

Comment:
We had a discussion about this, and explicitly decided _not_ to. Real FK constraints add a lot of cost to inserts, especially as giant tables become more giant. 
FK constraints are great, when you have relational data that changes regularly, since you don't want to delete a parent and leave a child stranded. But since events and annotations are never updated, there's little risk to leaving off costly FK constraints.

Reply:
I think the issue here, is the convenience of having a "default join" condition when using the query builder. The joins on event_annotation can be readily updated to have explicit "JOIN ON ..." statements, so that this Entity annotation isn't needed anymore, thankfully.

Reply:
I removed the FK/UC and specifically instructed TypeORM to _not_ create them.